### PR TITLE
SONARJNKNS-252 sonar scanner step pipeline

### PR DESCRIPTION
**Pull request by janos-ss**

### CODE REVIEW

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 1]
```diff
@@ -142,22 +151,24 @@ public String getSonarScannerName() {
...
+  @DataBoundSetter
   public void setSonarScannerName(String sonarScannerName) {
     this.sonarScannerName = sonarScannerName;
   }
 
   /**
    * Gets the JDK that this Sonar builder is configured with, or null.
    */
-  public JDK getJDK() {
+  public JDK getJdkFromJenkins() {
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: Can't we annotate @CheckForNull?

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 2]
```diff
@@ -214,12 +249,8 @@ public SonarRunnerInstallation getSonarRunnerInstallation() {
 
   @Override
   public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws InterruptedException, IOException {
-    performInternal(run, workspace, launcher, listener);
-  }
-
-  private boolean performInternal(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener) throws IOException, InterruptedException {
     if (!SonarInstallation.isValid(getInstallationName(), listener)) {
-      return false;
+      throw new AbortException("Invalid Sonar installation");
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: s/Sonar/SonarQube server

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 3]
```diff
@@ -109,9 +111,15 @@
...
+  private String task;
 
   @DataBoundConstructor
+  public SonarRunnerBuilder() {
+    // all fields are optional
+  }
+
+  // We're moving to using @DataBoundSetter instead and a much leaner @DataBoundConstructor
+  @Deprecated
   public SonarRunnerBuilder(String installationName, String sonarScannerName, String project, String properties, String javaOpts, String jdk, String task,
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: MAJOR Add the missing @deprecated Javadoc tag. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 4]
```diff
@@ -258,23 +288,21 @@ private boolean performInternal(Run<?, ?> run, FilePath workspace, Launcher laun
     env.put("SONAR_RUNNER_OPTS", getJavaOpts());
 
     long startTime = System.currentTimeMillis();
-    int r;
+    int exitCode;
     try {
-      r = executeSonarRunner(run, workspace, launcher, listener, args, env);
+      exitCode = executeSonarRunner(run, workspace, launcher, listener, args, env);
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: You can take the opportunity to rename the method: executeSonarQubeScanner() or executeSQScanner()

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 5]
```diff
@@ -258,23 +288,21 @@ private boolean performInternal(Run<?, ?> run, FilePath workspace, Launcher laun
...
     // if the analyis doesn't succeed, it will also be null
     SonarUtils.addBuildInfoTo(run, getSonarInstallation().getName());
-    return r == 0;
-  }
 
-  @Override
-  public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws IOException, InterruptedException {
-    return performInternal(build, build.getWorkspace(), launcher, listener);
+    if (exitCode != 0) {
+      throw new AbortException("Sonar runner exited with non-zero code: " + exitCode);
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: s/Sonar runner/SonarQube Scanner

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 6]
```diff
@@ -360,17 +388,19 @@ boolean populateConfiguration(ExtendedArgumentListBuilder args, Run<?, ?> build,
...
         }
         FilePath projectSettingsFilePath2 = workspace.child(projectSettingsFile);
         if (projectSettingsFilePath2.exists()) {
           // This must be what the user meant. Let it continue.
           projectSettingsFilePath = projectSettingsFilePath2;
         } else {
           // neither file exists. So this now really does look like an error.
-          listener.fatalError("Unable to find Sonar project settings at " + projectSettingsFilePath);
-          return false;
+          String msg = "Unable to find Sonar project settings at " + projectSettingsFilePath;
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: s/Sonar/SonarQube

**Comment by henryju** [comment]
Few comments then LGTM.

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java (hunk 1 above)
Comment (janos-ss) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: If we can trust the comment on the method, then definitely. I add it.

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 7]
```diff
@@ -289,7 +319,7 @@ private void handleErrors(Run<?, ?> build, TaskListener listener, SonarRunnerIns
     e.printStackTrace(listener.fatalError(errorMessage));
   }
 
-  private static int executeSonarRunner(Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, ArgumentListBuilder args, EnvVars env) throws IOException,
+  private static int executeSonarQubeScanner(Run<?, ?> build, FilePath workspace, Launcher launcher, TaskListener listener, ArgumentListBuilder args, EnvVars env) throws IOException,
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: MINOR Split this 182 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* MAJOR 1 major
* INFO 1 info

Watch the comments in this conversation to review them.

**Pull request merged**